src/alert/alert.tsx

```diff
@@ -19,8 +23,36 @@ export const Alert = ({
     actionLinkIcon,
     sizeType = "default",
     showIcon = false,
+    customAlertIcon,
+    maxHeight,
     ...otherProps
 }: AlertProps): JSX.Element => {
+    const [isShowMore, setIsShowMore] = useState(false);
+    const contentContainerHeightRef = useRef<HTMLDivElement>(null);
+    const [displayShowMore, setDisplayShowMore] = useState<boolean>(false);
+
+    // =============================================================================
+    // EVENT HANDLERS
+    // =============================================================================
+
+    useEffect(() => {
+        // forces line-clamp to trigger so that we can determine if content is collapsible
+        setIsShowMore(!maxHeight);
+        setDisplayShowMore(!!maxHeight);
```

Comment:
could you refactor this? I want to refrain from setting the state multiple times. Could you also extract this processing into a helper function